### DISCUSSION TITLE
scripting examples for some pics at tasmota page

### DISCUSSION THREAD

**Discussion by Sfre12** [arendst/Tasmota#17353] [Q&A] [answered]
Hey .
I found some new examples at the tasmota scripting page section..
https://tasmota.github.io/docs/Scripting-Language/#energy-collector .....and other engery....
there are also many more. I am very interessted to this pics and the script code which is in behind.
Is there any possibility to get this examples?

**Comment by gemu2015** [comment]
most of these require some experience with scripts and also the use of external script editor.
i will have to remove personal urls from the sources and add the compiler flags needed.
this will take some time. at least over the christmas holidays i will post the underlying scripts.

**Comment by gemu2015** [comment, marked as answer]
@Sfre12 
i added a folder "special_scripts" to the dropbox script editor download which contains most of these complex examples and also some live graphs

a few words to the scripts:
i wanted to get rid of my raspberries and also iobroker and do everything with ESPs
so all my devices communicate via global variables not MQTT (a mix of esp8266 and esp32)
a central ESP32 collects 7 meter devices and some other data and stores every 15 minutes on SD card
several devices with displays (epaper and others)  show essential  info throughout the house.
most of my devices also show graphs and extended data on their WEBUI.
one central ESP32 generates several graphics on demand, like daily data, weekly data, or monthly data.
it retrieves the data via special database commands from the data collectors SD card.
(which is done in multitasking script subroutines because it takes some time to extract the data from the database)

**Reply by Sfre12** [reply]
Hey Gemu.
Thanks a lot for doing this.
Since the last time I wrote about this, I've been doing some research on global variables. And, I succeeded. I have upgraded the firmware on all Tasmota devices accordingly and now the data exchange between the Tasmota devices works.
Well, the one with the graphical representation is now the next thing I'm going to do.
One more small thing. Unfortunately, I couldn't find the download link to Dropbox for "special_scripts". Can you put it in here again or tell me where I can find it

**Reply by gemu2015** [reply]
it is in the same link as for the script editor that you must use for these large scripts:

https://www.dropbox.com/sh/0us18ohui4c3k82/AACcVmpZ4AfpdrWE_MPFGmbma?dl=0
